src/helpers/getPathOr.js

```diff
@@ -0,0 +1,59 @@
+/** @license ISC License (c) copyright 2019 original and current authors */
+/** @author Ian Hofmann-Hicks */
+
+const curry = require('../core/curry')
+const isArray = require('../core/isArray')
+const isDefined = require('../core/isDefined')
+const isEmpty = require('../core/isEmpty')
+const isInteger = require('../core/isInteger')
+const isNil = require('../core/isNil')
+const isString = require('../core/isString')
+
+function fn(name) {
+  function getPathOr(def, keys, target) {
+    if(!isArray(keys)) {
+      throw new TypeError(
+        `${name}: Array of Non-empty Strings or Integers required for second argument`
+      )
+    }
+
+    if(isNil(target)) {
+      return def
+    }
+
+    let value = target
+    for(let i = 0; i < keys.length; i++) {
```

Comment:
Is the point of avoiding a recursive function here to not grow the stack?

Reply:
yep and for early exit.